cli/cmd/vuln_host.go

```diff
@@ -493,93 +510,141 @@ func hostVulnSummaryFromHostDetail(hostVulnSummary *api.HostVulnCveSummary) (str
+			item.packageStatus == s.packageStatus {
+			slice[i].cveCount++
+			if host {
+				slice[i].hostCount++
+			}
+			return slice
+		}
+	}
+	return append(slice, s)
+}
+
+func hostVulnPackagesTable(cves []api.HostVulnCVE, withHosts bool) ([][]string, string) {
+	var (
+		out                [][]string
+		filteredPackages   []packageTable
+		aggregatedPackages []packageTable
+	)
+
 	for _, cve := range cves {
 		for _, pkg := range cve.Packages {
-			// if the user wants to show only vulnerabilities of acive packages
+			pack := packageTable{
+				cveCount:       1,
+				severity:       strings.Title(pkg.Severity),
+				packageName:    pkg.Name,
+				currentVersion: pkg.Version,
+				fixVersion:     pkg.FixedVersion,
+				packageStatus:  pkg.PackageStatus,
+			}
+			if withHosts {
+				pack.hostCount = 1
+			}
+
 			if vulCmdState.Active && pkg.PackageStatus == "" {
+				filteredPackages = aggregatePackagesWithHosts(filteredPackages, pack, withHosts)
 				continue
 			}
+
 			if vulCmdState.Fixable && pkg.FixedVersion == "" {
+				filteredPackages = aggregatePackagesWithHosts(filteredPackages, pack, withHosts)
 				continue
 			}
 
-			added := false
-			for i := range out {
-				if out[i][1] == strings.Title(pkg.Severity) &&
-					out[i][2] == pkg.Name &&
-					out[i][3] == pkg.Version &&
-					out[i][4] == pkg.FixedVersion &&
-					out[i][5] == pkg.PackageStatus {
-
-					if countCVEs, err := strconv.Atoi(out[i][0]); err == nil {
-						out[i][0] = fmt.Sprintf("%d", (countCVEs + 1))
-						added = true
-					}
-
-					if withHosts {
-						if countHosts, err := strconv.Atoi(out[i][6]); err == nil {
-							prevCount := stringToInt(pkg.HostCount)
-							out[i][6] = fmt.Sprintf("%d", (countHosts + prevCount))
-							added = true
-						}
-					}
-
+			if vulCmdState.Severity != "" {
+				if filterSeverity(pkg.Severity, vulCmdState.Severity) {
+					filteredPackages = aggregatePackagesWithHosts(filteredPackages, pack, withHosts)
+					continue
 				}
 			}
+			aggregatedPackages = aggregatePackagesWithHosts(aggregatedPackages, pack, withHosts)
+		}
+	}
 
-			if added {
-				continue
-			}
-
-			row := []string{
-				"1",
-				strings.Title(pkg.Severity),
-				pkg.Name,
-				pkg.Version,
-				pkg.FixedVersion,
-				pkg.PackageStatus,
-			}
-			if withHosts {
-				row = append(row, pkg.HostCount)
-			}
-			out = append(out, row)
+	for _, p := range aggregatedPackages {
+		output := []string{
+			strconv.Itoa(p.cveCount),
+			p.severity,
+			p.packageName,
+			p.currentVersion,
+			p.fixVersion,
+			p.packageStatus}
+		if p.hostCount > 0 {
+			output = append(output, strconv.Itoa(p.hostCount))
 		}
+		out = append(out, output)
 	}
 
 	// order by severity
 	sort.Slice(out, func(i, j int) bool {
 		return severityOrder(out[i][1]) < severityOrder(out[j][1])
 	})
 
-	return out
+	if len(filteredPackages) > 0 {
+		filteredOutput := fmt.Sprintf("%v of %v package(s) showing \n", len(out), len(aggregatedPackages)+len(filteredPackages))
```

Comment:
nit: Everytime I read this footer I feel it looks way better if we type it like:
```
showing X of Y package(s)
```
But maybe this is just me..... 😓 
![tenor-83302290](https://user-images.githubusercontent.com/5712253/113948543-575f8d80-97ca-11eb-9ae0-304ce86105f2.gif)

Reply:
I quite like this way, with the most important details first. But open to be persuaded otherwise.

`showing 1 of 100`
vs
`1 of 100 showing`

Reply:
all good, this is just my bad English 😂

![tenor-168867785](https://user-images.githubusercontent.com/5712253/114110046-d40a6e80-9893-11eb-9dda-99981546f972.gif)